CONTRIBUTING.md

```diff
@@ -1,135 +1,384 @@
-# How to contribute to HDF5
+# HDF5 Development Guide
 
-The HDF Group encourages community members to contribute to the HDF5 project. We accept and are very grateful for any contributions,
-from minor typos and bug fixes to new features. The HDF Group is committed to work with the code contributors and make contribution
-process enjoyable and straightforward.
+Welcome to the HDF5 development community! This comprehensive guide covers everything you need to know
+about contributing to HDF5, from getting started to submitting your changes.
 
-This document describes guiding principles for the HDF5 code contributors and does not pretend to address any possible
-contribution. If in doubt, please do not hesitate to ask us for guidance.
-***Note that no contribution may be accepted unless the donor agrees with the HDF Group software license terms
-found in the LICENSE file in every branch's top source directory.***
+> [!IMPORTANT]
+> No contribution can be accepted unless the contributor agrees to the HDF Group's software license terms,
+  which can be found in the LICENSE file located in the top source directory of every branch.
 
+## Table of Contents
 
-> We will assume that you are familiar with `git` and `GitHub`.  If not, you may go through the GitHub tutorial found at
-[https://guides.github.com/activities/hello-world/](https://guides.github.com/activities/hello-world/).  This tutorial should only take
-around 10 minutes.
+- [Getting Started](#getting-started)
+- [Prerequisites](#prerequisites)
+- [Getting the Source Code](#getting-the-source-code)
+- [Building for Development](#building-for-development)
+- [Source Code Overview](#source-code-overview)
+- [Development Conventions](#development-conventions)
+- [Contributing Changes](#contributing-changes)
+- [Testing](#testing)
+- [Documentation](#documentation)
+- [Getting Help](#getting-help)
```

Comment:
It looks like the "Command-line tools" and "Checklist for Contributors" sections are missing from the TOC